tls/s2n_client_hello.c

```diff
@@ -200,27 +200,29 @@ int s2n_client_hello_recv(struct s2n_connection *conn)
         client_hello->extensions.size = extensions_length;
         client_hello->extensions.data = s2n_stuffer_raw_read(in, extensions_length);
         notnull_check(client_hello->extensions.data);
+    }
+
+    return 0;
+}
+
+static int s2n_process_client_hello(struct s2n_connection *conn)
+{
+    struct s2n_client_hello *client_hello = &conn->client_hello;
 
+    if (client_hello->extensions.size > 0) {
         GUARD(s2n_client_extensions_recv(conn, &client_hello->extensions));
     }
 
-    /* Mark the collected client hello as available when parsing over and before the client hello callback */
+    /* Mark the collected client hello as available */
     client_hello->parsed = 1;
```

Comment:
This flag controls access to the handle for the ```s2n_client_hello``` struct on the connection.
https://github.com/awslabs/s2n/blob/a46beb023ac9c52e982fd85b0ee3eaf9ba696835/tls/s2n_client_hello.c#L38-L44

The user application needs to obtain the handle to be able to call the ```s2_client_hello_get_``` accessor functions.
The accessor functions retrieve info stored on the ```s2n_client_hello``` struct. This struct is filled during the parsing of the client hello, so the functions should be available for use only after the client hello has been parsed. At the same time, we want the user application to be able to use them in the client hello callback.

This is why this flag needs to be set at the end of parsing *and before the client hello callback*.

Reply:
It makes sense now, I will move this to s2n_client_hello_recv and set it immediately after calling s2n_parse_client_hello and before client hello callback and I will restore the comment. Thanks.